website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md

```diff
@@ -224,8 +224,9 @@ Use a constraint file to ensure that Beam dependencies in the launch environment
 
 The launch environment translates the  pipeline graph into a [runner-independent representation](https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto). This process involves serializing (or pickling) the code of the transforms. The serialized content is deserialized on the workers. If the runtime worker environment significantly differs from the launch environment, runtime errors might occur for the following reasons:
 
+* The Apache Beam version  must match in the submission and runtime environments. Python major.minor versions must match as well. Otherwise, the pipeline might fail with errors like `Pipeline construction environment and pipeline runtime environment are not compatible`. On older SDK versions, the error might be reported as `SystemError: unknown opcode`.
```

Comment:
IIRC, if python versions are not same, it used to fail with dill errors. Is this the one dill output as error?

Reply:
we now have a better error, https://github.com/apache/beam/blob/5bf6fbf26708e06cc7b2fe286e400194890ffa49/sdks/python/apache_beam/runners/worker/bundle_processor.py#L851